### PR TITLE
V1 chat: group consecutive action/observation events

### PR DESCRIPTION
**Pull request by rbren**
Split out from `rb/megachange`. Introduces collapsible groups for runs of consecutive action/observation events in the V1 chat stream, plus follow-up tweaks built on that infrastructure.

- **Group consecutive action/observation events in V1 chat.** New `group-events.ts` folds runs of groupable action/observation events (excluding finish/think actions, planning/task-tracker observations, hooks, errors, and message events) into a single collapsible `EventGroup`. `EVENT_GROUP_MIN_SIZE` controls the threshold (currently 3). The header shows `EVENT_GROUP$ACTIONS_COMPLETED` once finished, or `EVENT_GROUP$ACTIONS_PROGRESS` plus the in-flight action's title while a member action has not yet been replaced by its observation.
- **Hoist agent thoughts out of action groups** so reasoning still surfaces above the collapsed run.
- **Group pairs of consecutive actions in the chat stream** (lower the threshold from 3 → 2 so even short runs collapse).
- **Hide inner left bar on events nested inside an EventGroup** via a new `IsInEventGroupContext` so the inner card border doesn't double up against the group's outer rail.

_This PR was created by an AI agent (OpenHands) on behalf of @rbren as part of splitting the `rb/megachange` branch into reviewable PRs._

### CODE REVIEW

**Comment by rbren** [comment]
@openhands address the review feedback

File: src/components/v1/chat/messages.tsx [hunk 1]
```diff
@@ -23,26 +28,60 @@ export const Messages: React.FC<MessagesProps> = React.memo(
     // This ensures only one preview per user message "phase"
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/v1/chat/messages.tsx:27]: Fixed in 0680640 by widening the memo guard to include both messages/allEvents lengths and their last event IDs, so action→observation swaps and allEvents updates trigger a re-render.

_This reply was generated by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/v1/chat/group-events.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  };
+
+  events.forEach((event, index) => {
+    if (isGroupableEvent(event)) {
+      const thoughtAction = getThoughtSourceAction(event, allEvents);
+      if (thoughtAction) {
+        flushRun();
+        items.push({ kind: "thought", action: thoughtAction, index });
+      }
+      if (!run) run = { events: [], startIndex: index };
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/v1/chat/group-events.ts:102]: Fixed in 0680640 by de-duplicating hoisted thoughts by source action ID in `groupEvents`, so a mixed `[action, observation]` UI array only emits a single thought item.

_This reply was generated by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/v1/chat/event-message-components/event-group.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    : t(I18nKey.EVENT_GROUP$ACTIONS_COMPLETED, { count: totalCount });
+
+  const Chevron = expanded ? ArrowUp : ArrowDown;
+
+  return (
+    <div
+      className="my-2 w-full border-l-2 border-neutral-300 pl-2 py-2 text-sm"
+      data-testid="event-group"
+    >
+      <button
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/v1/chat/event-message-components/event-group.tsx:68]: Fixed in 0680640 by wiring the toggle to the expanded body with `aria-controls`, `role="region"`, and `aria-labelledby`, plus regression coverage in `event-group.test.tsx`.

_This reply was generated by an AI agent (OpenHands) on behalf of @rbren._

File: __tests__/components/v1/chat/group-events.test.ts [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+  actionId: string,
+): ObservationEvent<PlanningFileEditorObservation> => ({
+  ...makeBashObservation(id, actionId),
+  observation: {
+    kind: "PlanningFileEditorObservation",
+    content: [{ type: "text", text: "plan" }],
+  } as PlanningFileEditorObservation,
+});
+
+describe("isGroupableEvent", () => {
```
[the remark this answers is not in the document]
Comment (rbren) [reply __tests__/components/v1/chat/group-events.test.ts:154]: Added the requested exclusions coverage in 0680640 for `TaskTrackerObservation`, `AgentErrorEvent`, and `HookExecutionEvent`.

_This reply was generated by an AI agent (OpenHands) on behalf of @rbren._

File: __tests__/components/v1/chat/event-message-components/event-group.test.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    content: [{ type: "text", text: "ok" }],
+    command,
+    exit_code: 0,
+    error: false,
+    timeout: false,
+    metadata: {} as never,
+  },
+});
+
+describe("EventGroup", () => {
```
[the remark this answers is not in the document]
Comment (rbren) [reply __tests__/components/v1/chat/event-message-components/event-group.test.tsx:66]: Added accessibility regression coverage in 0680640 for `aria-expanded`/`aria-label` toggling, the region wiring, and the empty-events edge case.

_This reply was generated by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/v1/chat/group-events.ts [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+ * `RenderedItem`, the current run is flushed, and a new run is started with
+ * the event itself. This keeps reasoning text in the main message stream
+ * instead of buried inside a collapsed action group.
+ *
+ * `allEvents` should be the full event history so observations can find
+ * their matching action; if omitted, it defaults to `events`.
+ */
+export const groupEvents = (
+  events: OpenHandsEvent[],
+  minSize: number = EVENT_GROUP_MIN_SIZE,
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/v1/chat/group-events.ts:67]: Handled in 0680640 by validating `minSize >= 1` inside `groupEvents`, with a matching regression test.

_This reply was generated by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/v1/chat/event-message-components/event-group.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+export function EventGroup({ events, children }: EventGroupProps) {
+  const { t } = useTranslation("openhands");
+  const [expanded, setExpanded] = React.useState(false);
+
+  // Each ObservationEvent in the group is a completed action.
+  // An ActionEvent that's still here (i.e. not yet replaced by its observation
+  // in the UI events array) is the action currently in flight.
+  const pendingAction = events.find((e): e is ActionEvent => isActionEvent(e));
+  const completedCount = events.filter(isObservationEvent).length;
+  const totalCount = events.length;
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/v1/chat/event-message-components/event-group.tsx:47]: Handled in 0680640 by returning `null` for empty event groups, and covered it with a component test.

_This reply was generated by an AI agent (OpenHands) on behalf of @rbren._
